### PR TITLE
TST: Actually process the generators.

### PR DESCRIPTION
**Pull request by ericdill**
Actually prcessing the generator uncovered a few bugs! Now I need to 
squash them.

This PR will close #78 when done.

### CODE REVIEW

File: databroker/tests/test_broker.py [hunk 1]
```diff
@@ -63,20 +63,27 @@ def test_basic_usage():
...
 
     # test time shift issue GH9
     table = get_table(db[105])
     assert_true(table.notnull().all().all())
 
+
+def test_get_events_bad_key():
+    hdr = db[-1]
+
+    list(get_events(hdr, fields=['abcd123']))
```
Comment (danielballan) [inline databroker/tests/test_broker.py]: This should assert that it raises, right?

Comment (ericdill) [inline databroker/tests/test_broker.py]: yes, good catch.  I was having trouble making it raise and wanted to make sure it was actually failing!

Comment (ericdill) [inline databroker/tests/test_broker.py]: I just wrapped the function in an raises decorator

**Comment by danielballan** [comment]
Master is currently broken. Merging this (which adds tests that expose the problems) merely makes that official. Merging.

**Comment by danielballan** [comment]
Fixes for both of these are already in #32

**Pull request merged**